eth/vm/logic/arithmetic.py

```diff
@@ -205,3 +205,18 @@ def shr(computation):
         result = (value >> shift_length) & constants.UINT_256_MAX
 
     computation.stack_push(result)
+
+
+def sar(computation):
+    """
+    Arithmetic bitwise right shift
+    """
+    shift_length, value = computation.stack_pop(num_items=2, type_hint=constants.UINT256)
+    value = unsigned_to_signed(value)
+
+    if shift_length >= 256:
+        result = 0 if value >= 0 else signed_to_unsigned(-1)
```

Comment:
Can we *constantify* `signed_to_unsigned(-1)`

Reply:
Makes sense! However, I'm not sure if I got the naming right because my bits and bytes fu is pretty bad.

I [added](https://github.com/ethereum/py-evm/pull/1134/files#diff-1a697f3a2088726b12d5e69c8fee879eR16) the following constant to `eth.constants`

```py
UINT_255_NEGATIVE_ONE = -1 + UINT_256_CEILING
```
My understanding is that this is representing the number `-1` according to the rules of the [two's complement](https://en.wikipedia.org/wiki/Two's_complement) which uses the most significant bit to represent the sign and leaving us with 255 other bits to represent the number.

Btw, the constant basically inlines what `signed_to_unsigned` does because I can't import it here (circular import)

Reply:
@pipermerriam interpreted :+1: as indicator that the constant name seems reasonable and merged it